sdk/include/crt/xmmintrin.h

```diff
@@ -1,25 +1,42 @@
-/**
- * This file has no copyright assigned and is placed in the Public Domain.
- * This file is part of the w64 mingw-runtime package.
- * No warranty is given; refer to the file DISCLAIMER within this package.
+/*
+ * xmmintrin.h
+ *
+ * This file is part of the ReactOS CRT package.
+ *
+ * Contributors:
+ *   Timo Kreuzer (timo.kreuzer@reactos.org)
+ *
+ * THIS SOFTWARE IS NOT COPYRIGHTED
+ *
+ * This source code is offered for use in the public domain. You may
+ * use, modify or distribute it freely.
+ *
+ * This code is distributed in the hope that it will be useful but
+ * WITHOUT ANY WARRANTY. ALL WARRANTIES, EXPRESS OR IMPLIED ARE HEREBY
+ * DISCLAIMED. This includes but is not limited to warranties of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
  */
 
 #pragma once
 #ifndef _INCLUDED_MM2
 #define _INCLUDED_MM2
 
-#include <crtdefs.h>
 #include <mmintrin.h>
 
-#ifdef __clang__
+#if defined(_MM2_FUNCTIONALITY) && !defined(_MM_FUNCTIONALITY)
+#define _MM_FUNCTIONALITY
+#endif
 
-typedef        float __v4sf __attribute__((__vector_size__(16)));
-typedef   signed int __v4si __attribute__((__vector_size__(16)));
-typedef unsigned int __v4su __attribute__((__vector_size__(16)));
+#if !defined _VCRT_BUILD && !defined _INC_MALLOC
+//#include <malloc.h> // FIXME: This breaks build
```

Comment:
Why the malloc inclusion? Is it for these `__mm_malloc` and free down the file? (also I see those latter are defined only when `__ICL` is defined; maybe worth adding that here too? Or you didn't do this in order to have the header compatible with an equivalent one from MSVC or another compiler?)

Reply:
The malloc inclusion is what MS headers include. I don't see any actual reason for it, other than "his is what Windows does".
But in this case it's what public headers do, so I think it's worth to try to be conforming.